sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2421,6 +2423,60 @@ multiply_with_overflow_check(T x, T y) {
     return x * y;
 }
 
+namespace matrix_ext = ext::oneapi::experimental::matrix;
+
+// Matrix type string to matrix_type enum value conversion
+// Note: matrix type strings are defined in template specialization for
+// convertTypeToMatrixTypeString above
+std::optional<matrix_ext::matrix_type>
+convertMatrixTypeStringMatrixTypeEnumValue(std::string MatrixTypeString) {
+  assert(!MatrixTypeString.empty() &&
+         "MatrixTypeString type string can't be empty. Check if required "
+         "template specialization for convertTypeToMatrixTypeString exists.");
+  if ("matrix_type::bf16" == MatrixTypeString)
```

Comment:
This looks like a lot of unnecessary repeated comparisons with "matrix_type::".

I would suggest to maybe accept `string_view` in here, `assert` that it has the right prefix. Then use [`remove_prefix`](https://en.cppreference.com/w/cpp/string/basic_string_view/remove_prefix) to ignore it and compare remaining symbols only.

Using `startswith` and `endswith` would be even better, but that is C++20, unfortunately